.github/workflows/lint-testdata.yml

```diff
@@ -0,0 +1,85 @@
+name: Testdata Lint Check
+
+# Trigger the workflow on pull requests and direct pushes to any branch
+on:
+  push:
+  pull_request:
+
+jobs:
+  check-lint-for-project-v2:
+    uses: kubernetes-sigs/kubebuilder/.github/workflows/lint-testdata.yml@master
+    with:
+      name: project-v2
+      working-directory: testdata/project-v2
+
+  check-lint-for-project-v3:
```

Comment:
I think we can test only for go/v4 and no longer care about v3.
We are close to make the go/v4 stable and make the default scaffold which would mean
deprecate v3.